apm-agent-tracer/src/main/java/co/elastic/apm/agent/tracer/AbstractSpan.java

```diff
@@ -184,4 +184,17 @@
     void incrementReferences();
 
     void decrementReferences();
+
+    /**
+     * @return {@literal true} when span limit is reached and the caller can optimize and not create a span. The caller
+     * is expected to call this method before every span creation operation for proper dropped spans accounting. If not
+     * called before attempting span creation, a span will be created and dropped before reporting.
+     * <br/>
+     * Expected caller behavior depends on the returned value:
+     * <ul>
+     *     <li>{@literal true} returned means the caller is expected to NOT call {@link #createSpan()} or {@link #createExitSpan()}</li>
+     *     <li>{@literal false} returned means the caller MAY call {@link #createSpan()} or {@link #createExitSpan()}</li>
+     * </ul>
+     */
+    boolean shouldSkipChildSpanCreation();
```

Comment:
That's exactly why I took this simple approach here, and it would have been quite consistent with `createExitSpan` that is expected to return `null` in case there is already an active exit span.

Another alternative would be to have a way to return a no-op span, which could avoid most of the refactoring but could still have some unexpected side-effects.

I think it could be worth investing a bit of time for a refactoring/tech-debt PR.